OneSignal/OneSignal-Android-SDK
# [User Model] Rename IAM event listeners

# Description
## One Line Summary
Rename IAM event listeners based on new naming convention.

## Details
* Renamed all "handlers" to be either a "listener" or an "observer" depending on their purpose. All callbacks now have pattern "onXXXXXX" for each callback.
* Make all notification listeners 0:* (some were 0:1)
* Use the terminology "click" instead of "open".
* Update example application for new naming conventions and patterns.

### Motivation
A desire to align terminologies and naming conventions across the OneSignal SDKs.

### Scope
This change is primarily a name change within the IAM module/namespace. No functional changes were made.

### OPTIONAL - Other
**OPTIONAL -** Feel free to add any other sections or sub-sections that can explain your PR better.

# Testing

## Manual testing
* Tested IAM lifecycle and click handle listeners through example app to ensure they are still being called.

# Affected code checklist

# Checklist
## Overview
     - If it is hard to explain how any codes changes are related to each other then it most likely needs to be more than one PR

## Testing

## Final pass
      - Simplify with less code, followed by splitting up code into well named functions and variables, followed by adding comments to the code.
      - WIP (Work In Progress) is ok, but explain what is still in progress and what you would like feedback on. Start the PR title with "WIP" to indicate this.


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/OneSignal/OneSignal-Android-SDK/1766)
